### PR TITLE
Feature/language agnostic comp sci

### PR DESCRIPTION
**Pull request by JonColman94**
First edition of the Language Agnostic Computer Science Course described [in this issue](https://github.com/TheOdinProject/curriculum/issues/15979)

Lessons are found in the /computer_science directory.

The course discusses Pseudocode, Recursion and it's comparisons with Iteration; some different types of sort (Insertion Sort, Merge Sort, Quicksort and Bogosort), as well as search algorithms (Linear and Binary Search). Discussion on Big O notation, with examples of what some O notations look like in code ( O(1), O(n), O(n^2) and O(logn) ) specifically. As well as putting our legacy Linked List and Tree projects to good use.

Extended tree discussion to briefly cover Depth-Limited Search and Iterative Deepening.

New Projects include comparing Recursive solutions with Iterative solutions; and putting practical understanding of Big O notation to code. 

I do believe this course will be in need of further refinement; but I'd like to at least have some review on this first draft.

Placed the original Ruby Comp Sci lessons in a /legacy directory inside the /computer_science directory for sake of reference.

### CODE REVIEW

**Comment by KevinMulhern** [comment]
Hey @Jan-Kotze, thanks for doing this. Sorry it hasn't been reviewed yet. We're wrapping up a couple of big projects and will get to this shortly.

**Comment by JonColman94** [comment]
No rush, take all the time you need.

File: computer_science/Legacy/lesson_a_very_brief_intro_to_cs.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+### Learning Outcomes
+Look through these now and then use them to test yourself after doing the assignment:
+
+* What is an algorithm?
+* What is pseudo-code?
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+  1. Skim this [Introduction to Algorithms by David Malan](http://ed.ted.com/lessons/your-brain-can-solve-algorithms-david-j-malan) on TedEd to see how to think about algorithms.
```
Comment (Deetss) [inline computer_science/Legacy/lesson_a_very_brief_intro_to_cs.md:22]: This link is dead

**Comment by CouchofTomato** [comment]
Hey @JonColman94 

Just want to offer my apologies this hasn't been picked up sooner. We really do appreciate your undertaking and I'm going to review it this week.

File: computer_science/lesson_maps_stacks_and_queues.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+An alternative name for a `queue` is a `FIFO (First-In-First-Out)` queue.
+
+Some languages have builtin queues ready for use; but Ruby and Javascript do not explicitly have `queues` or either of the mentioned functions. However, both languages [Ruby](https://ruby-doc.org/core-2.6/Array.html#method-i-shift) and [Javascript](https://devdocs.io/javascript/global_objects/array/shift) have the `#shift` method which, when used with `#push`; allows the easy implementation of `queues`.
+
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+  1. Use pop, push and shift in your language of choice and get a feel for how these work! If you're not using Ruby or Javascript for this course, you might need to google around to find Queue implementation.
```
Comment (CouchofTomato) [inline computer_science/lesson_maps_stacks_and_queues.md:78]: Why would they not be using JS or Ruby?

File: computer_science/lesson_the_big_o.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+First, lets get into a bit of math (let `joy` be `unconfined`). Say you written some function and at worst it takes very specific number of operations to tackle a problem of size n, for example:
+
+operations for input n : 6n^2 + 300n + 7
+
+If we were to make n ***very big***, our 6n^2 grows ***much*** larger than our 300n or our 7. So for really big n...
+
+operations for input for very big n ~ 6n^2
+
+This means that our operations is approximately 6n^2. But we're not quite done yet. Say we grow n by 10 times as much; we only grow the number of operations by 100. The 6 in our little equation above is irrelevant. So we also ignore the six, and we're left with what we call our `Big O`, our worst-case growth with input n:
```
Comment (CouchofTomato) [inline computer_science/lesson_the_big_o.md:29]: Should be operations are, or operation is

File: computer_science/lesson_the_big_o.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+However, it should be noted that for *most* cases, `Big O notation` is an accurate quantifier for performance on a large scale. There exists some cases where although an algorithm *has* a more efficient `Big O`, in reality that efficiency might only come in for `impossibly big` data sets. So care should still be made to properly research an algorithm if you intend on employing it.
+
+It should also be noted that `Big O` is theoretical. Again, it should mostly hold in the real world; but there are other tricks you will learn throughout your career to further optimise your code if necessary.
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+  1. Have another look at the algorithms mentioned in [our previous lesson](http://localhost:3000/courses/computer-science/lessons/a-general-view-on-algorithms) What are their Big O notations? What do these notations mean?
```
Comment (CouchofTomato) [inline computer_science/lesson_the_big_o.md:146]: I think we should find at least one good resource which also covers big-0

**Review by CouchofTomato — CHANGES_REQUESTED** [review]
Overall this is an amazing effort. Just a few things we need to consider around the assignment resources.

**Pull request merged**